### PR TITLE
Add glide app module for configuration

### PR DESCRIPTION
**Pull request by thornbill**
**Changes**
Adds an app module for Glide to manage common configuration options and manage log level

**Issues**
N/A

### CODE REVIEW

File: app/src/main/java/org/jellyfin/androidtv/JellyfinGlideModule.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	override fun applyOptions(context: Context, builder: GlideBuilder) {
+		builder.setDefaultRequestOptions(
+			// Set default disk cache strategy
+			RequestOptions()
+				.diskCacheStrategy(DiskCacheStrategy.RESOURCE)
+		)
+
+		// Silence image load errors
+		builder.setLogLevel(Log.ERROR)
+	}
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/JellyfinGlideModule.kt:22]:
```suggestion
	override fun applyOptions(context: Context, builder: GlideBuilder) = with(builder) {
		setDefaultRequestOptions(
			// Set default disk cache strategy
			RequestOptions().diskCacheStrategy(DiskCacheStrategy.RESOURCE)
		)

		// Silence image load errors
		setLogLevel(Log.ERROR)
	}
```

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

**Comment by linetrimmer** [comment]
I think usages of `Glide.with` need to be replaced with `GlideApp.with` to actually use the custom module.

**Comment by nielsvanvelzen** [comment]
> I think usages of `Glide.with` need to be replaced with `GlideApp.with` to actually use the custom module.

No I think that's what the annotation processor does, is it a horrible way of doing it? Yes

https://bumptech.github.io/glide/doc/configuration.html#applications

**Comment by linetrimmer** [comment]
>No I think that's what the annotation processor does, is it a horrible way of doing it? Yes
 https://bumptech.github.io/glide/doc/configuration.html#applications

The Flickr sample app linked in this section also uses `GlideApp.with()` instead of `Glide.with()`.

I was looking at https://bumptech.github.io/glide/doc/generatedapi.html#using-the-generated-api where it says

> Applications can use the API by starting all loads with GlideApp.with() instead of Glide.with()

My bad if it's not necessary for this, though.

File: app/src/main/java/org/jellyfin/androidtv/JellyfinGlideModule.kt (hunk 1 above)
Comment (thornbill) [reply app/src/main/java/org/jellyfin/androidtv/JellyfinGlideModule.kt]: I had to add a `: Unit` in there, but should be good now.

**Review by nielsvanvelzen — APPROVED** [review]
Can you confirm using just `Glide.xxx` instead of `JellyfinGlideMoldule.xxx` works?

**Comment by thornbill** [comment]
It is definitely honoring the log configuration when just using `Glide`. It may be a bit harder to verify the cache strategy is used. :wink:

**Comment by nielsvanvelzen** [comment]
If the logging works it can pick up the config so the caching should work :p

**Pull request merged**